policy/examples/tiered-parking-fees-total.json

```diff
@@ -0,0 +1,57 @@
+{
+  "updated": 0,
+  "version": "1.2.0",
+  "data": {
+    "policy": [
+      {
+        "name": "Tiered Dwell Time Example",
+        "description": "If parked for <1hr $2 upon exit, if parked for 1-2 hours $4 upon exit, if parked for longer than 2 hours $10 upon exit",
+        "policy_id": "2800cd0a-7827-4110-9713-b9e5bf29e9a1",
+        "start_date": 1558389669540,
+        "publish_date": 1558389669540,
+        "end_date": null,
+        "prev_policies": null,
+        "provider_ids": [],
+        "currency": "USD",
+        "rules": [
```

Comment:
I want to make sure I understand the process for interpreting these rules, so I'm going to walk through an example. Let's say we have a 1.25 hour long parking event. The total charge for that under this policy would be $4.

So the excruciatingly detailed process for calculating that would be:

1. Iterate over rules
1. Looking at the first rule:
    1. This is a `once_on_unmatch` rule so I need to figure out if I have parking event that has gone from matching this rule to not matching this rule
    1. This rule has `rule_units` of hours and I normalize my event duration with the rule's units
    1. The rule has a `maximum` of 2, which is more than the duration of my event, so I move on to the next rule
1. Looking at the second rule:
    1. This is a `once_on_unmatch` rule so I need to figure out if I have parking event that has gone from matching this rule to not matching this rule
    1. This rule has `rule_units` of hours and I normalize my event duration with the rule's units
    1. This rule has a `maximum` of 1. **A `once_on_unmatch` rule always applies when my event is longer than a rule's `maximum`**, so I need to use this rule to calculate a fee.
    1. The fee for this rule is 400 = $4
    1. **`once_on_unmatch` rules always apply to the entirety of a parking event** and **once I've applied rules to my entire event I can stop iterating over rules, even if there are more**, so I break out of the loop over rules.
        - Is this true? Could there be more applicable rules, like rate rules? Or would those be in a different policy?
1. I sum up the charges for each rule for a total of $4

Does that all seem right?

Reply:
All sounds right to me! Short-circuiting should behave exactly how you described.